tICA/scripts/feature_helpers/brain_masks.m

```diff
@@ -0,0 +1,151 @@
+function brain_masks(StudyFolder, GroupAverageName, Resolution, OutputfMRIName, MaskSavePath, ConfigFilePath, Dilate)
+%Generate cifti masks for spatial tICA features. There are two categories of masks, 1. specific brain region masks, including CSF, WM, GM and
+%subcortical area, 2. brain boundary masks
+%   Detailed explanation goes here
+switch nargin
+  case 0
+    error('Must feed arguments!!!') 
+  case 1
+    error('Must specify group-average name!!!') 
+  case 2
+    error('Must specify resolution!!!') 
+  case 3
+    error('Must specify output fMRI name!!!') 
+  case 4
+    MaskSavePath = [];
+    ConfigFilePath = [];
+    Dilate=[];
+  case 5
+    ConfigFilePath = [];
+    Dilate=[];
+  case 6
+    Dilate=[];
+  case 7
+  otherwise
+    error('<=7 inputs are accepted.')
+end
+
+if isempty(ConfigFilePath)
+  ConfigFilePath = '/media/2TBB/Connectome_Project/Pipelines/global/config';
+end
+if isempty(MaskSavePath)
+  MaskSavePath = [StudyFolder '/' GroupAverageName '/MNINonLinear/ROIs'];
+end
+VolumeTemplate = [GroupAverageName '_CIFTIVolumeTemplate_' OutputfMRIName '.' Resolution '.dscalar.nii'];
+
+if isempty(Dilate)
+  Dilate = '5.0';
+end
+
+resolution=str2double(Resolution);
+erode_mm=cell(1,5);
+for i=1:5
+    erode_mm{i}=num2str(resolution*i);
```

Comment:
Exact multiples of the voxel resolution are actually slightly unstable as dilation distances in wb_command (particularly for binary masks), because rounding error determines whether or not the "cardinal-direction" voxel gets included (whether the distance rounds to 3.9999 or 4.0001 matters if your distance limit is exactly 4, while `1.9 * resoution` or `2.1 * resolution` would be guaranteed to either exclude or include the "tip" voxels of the sphere, respectively).  Half-voxel distances ([1.5, 2.5, 3.5] * resolution) might actually seem the most spherical in terms of the voxels that get included, as it naturally matches "distance to the far edge of this voxel".

Reply:
Also, what is the purpose of eroding based on number of voxels, rather than by some anatomically relevant distance?  What are these eroded masks used for?

Reply:
These eroded masks are used to obtain brain boundary features (like mean, std, etc.). For generalization purpose on both 3T(2mm) and 7T(1.60mm) dataset, or even other HCP-style data(2.4mm), I use the resolution to control the range of these masks.